opal-core-ws/src/main/java/org/obiba/opal/web/project/ProjectTransientDatasourcesResource.java

```diff
@@ -80,10 +96,66 @@ public Response createDatasource(Magma.DatasourceFactoryDto factoryDto) {
           .entity(Dtos.asDto(ds).build()).build();
     } catch(MagmaRuntimeException e) {
       MagmaEngine.get().removeTransientDatasource(uid);
+
+      if (e instanceof DatasourceParsingException) {
+        cacheDatarsourceParseErrorLog((DatasourceParsingException) e);
+      }
+
       throw e;
     }
   }
 
+  private void cacheDatarsourceParseErrorLog(DatasourceParsingException de) {
+    StringBuilder log = new StringBuilder();
+
+    for(DatasourceParsingException c : de.getChildrenAsList()) {
+      Object[] args = c.getParameters().toArray();
+
+      if (args.length > 2)
+        log.append(String.format("[%s: %s] ", Arrays.copyOfRange(args, 0, 2)));
+
+      log.append(c.getMessage());
+
+      if (args.length == 5)
+        log.append(String.format("(table: %s, variable: %s, category: %s)", Arrays.copyOfRange(args, 2, args.length)));
+      else if (args.length == 4)
+        log.append(String.format("(table: %s, variable: %s)", Arrays.copyOfRange(args, 2, args.length)));
+
+      log.append('\n');
+    }
+
+    Serializable sessionId = getSessionId();
+    if(sessionId != null)
+      getCache().put(new Element(sessionId, log.toString()));
+  }
+
+  private Serializable getSessionId() {
+    return SecurityUtils.getSubject().getSession().getId();
+  }
+
+  private Cache getCache() {
+    return cacheManager.getCache("opal-datasource-parse-error-log");
+  }
+
+  @GET
+  @Path("/opal-datasource-parse-error-log")
```

Comment:
Would prefer to rename as: /_last-errors

This way it won't commit too much on the type of errors:
GET /project/{name}/transient-datasources/_last-errors